src/include/network/lan/address.rb

```diff
@@ -1436,6 +1441,19 @@ def initialize_address_settings
       @settings["BOOTPROTO"] = "static" if LanItems.operation == :add && @force_static_ip
     end
 
+    # Return wether the {Yast:Routing} devices list needs to be updated or not
+    # to include the current interface name
+    #
+    # @return [Boolean] false if the current interface name is already present
+    def update_routing_devices?
+      !Routing.devices.include?(LanItems.current_name)
+    end
+
+    # Convenience method to update the {Yast::Routing} devices list
+    def update_routing_devices!
```

Comment:
@teclator, just an idea, without having too much knowledge about this.

What about to move this two methods to `Routing` module, I mean

```ruby
class Routing # actually is something like class RotuinClass < Module ...
  ...

  def update_routing_devices(device_names) # or def update_devices(names)
    return if devices.include?(device_names)

    SetDevices(device_names)
  end
end
```

Then, you can remove those convenience methods and use `Routing.update_routing_devices` or just `Routing.update_devices` instead.

Reply:
Yep, I was already evaluating that.

Reply:
Mmm doesn't look so different than calling SetDevices(...)

Reply:
Right. It only avoids to call `Set` when not necessary :)